hash_practice/exercises.py

```diff
@@ -2,19 +2,44 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
+        Time Complexity: O(n)
         Space Complexity: ?
     """
-    pass
+    anagrams = {}
+    for word in strings:
+        sorted_word= ''.join(sorted(word))
+        if sorted_word not in anagrams:
+            anagrams[sorted_word] = [word]
+        else:
+            anagrams[sorted_word].append(word)
+    list_of_lists = []
+    for item in anagrams.values():
+        list_of_lists.append(item)
+    return list_of_lists
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    "e.g.: Input: nums = [1,1,1,2,2,3], k = 2, Output: [1,2]"
+    
+    dict={}
+    list=[]

```

Comment:
Two minor style nitpicks:

1. Ideally, I recommend picking more descriptive names for your variables rather than just what their type is. Especially in a language like Python the _reason_ for a variable is more important that the _type_. I would probably use a name like `num_frequencies` for `dict` and `result_list` for `list`. In a simple  problem like this, it's not a big deal, but picking good names, though challenging, can really make the difference between code that's easy to work with and code that's hard to work with. There's other places where you use very terse names, so I won't point out every one.
2. While you define `list` here on line 30, it doesn't really get used until line 40 below. I generally try to postpone variable definitions until the last possible moment, right before I need them or set them. This is mainly because it then makes it clearer the tie between a variable and the code that initializes or uses it.

Both of these are very minor, so no points lost.